cpp/src/arrow/ipc/test-common.h

```diff
@@ -223,15 +223,17 @@ Status MakeRandomBinaryArray(int64_t length, bool include_nulls, MemoryPool* poo
     if (include_nulls && values_index == 0) {
       RETURN_NOT_OK(builder.AppendNull());
     } else {
-      const std::string& value = values[values_index];
+      const std::string value =
+          i < int64_t(values.size()) ? values[values_index] : std::to_string(i);
```

Comment:
Why did you need this change? Is this just a debugging leftover?

Reply:
Not knowing the history of this particular function, I don't know what would be "better" in there. For my test I pretty much just want the consecutive numbers, otherwise it is very difficult to see what has gone wrong (I like the tests to give me the answer to that question if possible). I made a change to add a second function that implements that, but turns out this function is only used from MakeStringTypesRecordBatch, so we can't really have two implementations.